conversation.go

```diff
@@ -0,0 +1,131 @@
+		Display    bool   `json:"display"`
+		Name       string `json:"name"`
+	} `json:"created_via"`
+	CreatedBy ByLine `json:"created_by"`
+	CreatedOn Time   `json:"created_on"`
+}
+
+// ConversationSelector can modify the scope of a conversations lookup request - see WithLimit and WithOffset for examples.
+type ConversationSelector func(uri *url.URL)
+
+// GetConversation returns all conversations that the client has access to (max 200). Use WithLimit and WithOffset
+// to do pagination if that is what you want.
+func (client *Client) GetConversations(withOpts ...ConversationSelector) ([]Conversation, error) {
+	u, err := url.Parse("/conversation/")
+	if err != nil { // should never happen
+		return nil, err
+	}
+	for _, selector := range withOpts {
+		selector(u)
+	}
+
+	convs := []Conversation{}
+	err = client.Request("GET", u.RequestURI(), nil, nil, &convs)
+	return convs, err
+}
+
+// GetConversationEvents returns all events for the conversation with id conversationId. WithLimit and WithOffset can be used to do
+// pagination.
+func (client *Client) GetConversationEvents(conversationId uint, withOpts ...ConversationSelector) ([]ConversationEvent, error) {
+	u, err := url.Parse(fmt.Sprintf("/conversation/%d/event", conversationId))
+	if err != nil { // should never happen
+		return nil, err
+	}
+	for _, selector := range withOpts {
+		selector(u)
+	}
+
+	convs := []ConversationEvent{}
+	err = client.Request("GET", u.RequestURI(), nil, nil, &convs)
+	return convs, err
+}
+
+// Reply sends a (string) message to the conversation identified by conversationId. Only text strings are supported (that is
+// no embedding for now).
+func (client *Client) Reply(conversationId uint, reply string) (ConversationEvent, error) {
+	path := fmt.Sprintf("/conversation/%d/reply/v2", conversationId)
+	out := ConversationEvent{}
+
+	buf, err := json.Marshal(struct {
+		Text string `json:"text"`
+	}{Text: reply})
+	if err != nil {
+		return out, err
+	}
+	err = client.Request("POST", path, map[string]string{"content-type": "application/json"}, bytes.NewReader(buf), &out)
+	return out, err
+}
+
+// WithLimit sets a limit on the returned list of Conversations or ConversationEvents. limit must be in the range (0-200].
+func WithLimit(limit uint) ConversationSelector {
+	f := func(u *url.URL) {
+		q := u.Query()
+		q.Add("limit", strconv.Itoa(int(limit)))
+		u.RawQuery = q.Encode()
+	}
+	return ConversationSelector(f)
+}
+
+// WithOffset introduces an offset in the returned list of Conversations or ConversationsEvents.
+func WithOffset(offset uint) ConversationSelector {
+	f := func(u *url.URL) {
+		q := u.Query()
+		q.Add("offset", strconv.Itoa(int(offset)))
+		u.RawQuery = q.Encode()
+	}
+	return ConversationSelector(f)
+}
+
+// Unread manipulates the conversation request to only conversations with unread messages.
+func Unread() ConversationSelector {
+	f := func(u *url.URL) {
+		u.Path = path.Join(u.Path, "unread")
+	}
+	return ConversationSelector(f)
+}
```

Comment:
I think it could be good to split out some of these areas. Maybe the `podio` package should just be a client that can sign requests and `file`, `conversation` and  other packages have simple clients that in turn generate requests (these clients could then embed `podio.Client` to ease access). But maybe it would just be a little cumbersome to manually have to initialise each area as you need it. 

I'll try to create a `conversation` package and see if it works out.